### PR TITLE
Resolve some chplcheck warnings in modules

### PR DESCRIPTION
**Pull request by jabraham17**
This PR uses the new `--fixit` functionality of `chplcheck` to auto-detect and auto-fix code to be Chapeltastic. All of the code was checked by hand after the fixits were applied to ensure only correct code was emitted.

Testing
- [x] paratest
- [x] paratest with gasnet

[Reviewed by @DanilaFe]

### CODE REVIEW

File: modules/packages/LinearAlgebra.chpl [hunk 1]
```diff
@@ -3450,7 +3450,7 @@ module Sparse {
         end = D.shape(0);
       }
       var indices : [start..end] (D.idxType, D.idxType);
-      forall ind in {start..end} with (ref indices) {
```
Comment (DanilaFe) [inline modules/packages/LinearAlgebra.chpl:3453]: I wonder if this is a performance improvement. Do we track LinearAlgebra perf anywhere?

File: modules/standard/ChplConfig.chpl [hunk 2]
```diff
@@ -129,7 +129,7 @@ module ChplConfig {
   CHPL_MEM = __primitive("get compiler variable", "CHPL_MEM");
 
   /* See :ref:`readme-chplenv.CHPL_MAKE` for more information. */
-  @unstable("'ChplConfig.CHPL_MAKE' is unstable and may be replaced with a different way to access this information in the future");
```
Comment (DanilaFe) [inline modules/standard/ChplConfig.chpl:132]: Huh... Did this make `CHPL_MAKE` unstable when it wasn't before? If `;` here is the empty statement, then the attribute won't affect `CHPL_MAKE` prior to this PR. Is there a corresponding `test/unstable` test? If not, might be worth adding one.

File: modules/standard/List.chpl [hunk 3]
```diff
@@ -151,7 +151,7 @@ module List {
     type eltType;
 
     /*If `true`, this list will perform parallel safe operations.*/
-    @unstable("'list.parSafe' is unstable and is expected to be replaced by a separate list type in the future");
+    @unstable("'list.parSafe' is unstable and is expected to be replaced by a separate list type in the future")
     param parSafe = false;
```
Comment (DanilaFe) [inline modules/standard/List.chpl:155]: Ditto here about `;` on `parSafe`.

**Review by DanilaFe — APPROVED** [review]

File: modules/standard/ChplConfig.chpl (hunk 2 above)
Comment (jabraham17) [reply modules/standard/ChplConfig.chpl:132]: There is not a test of this, but `CHPL_MAKE` was still being marked as unstable before this as well

File: modules/packages/LinearAlgebra.chpl (hunk 1 above)
Comment (jabraham17) [reply modules/packages/LinearAlgebra.chpl:3453]: I don't think we have a lot, if any.

FWIW, I would not expect a major boost here. `{...}` has been shown to inhibit vectorization, but given the loop body here I doubt it matters much

File: modules/standard/List.chpl (hunk 3 above)
Comment (jabraham17) [reply modules/standard/List.chpl:155]: This one I know we have tests of, the `;` had no affect other than being stylistically wrong

I suppose we could argue that attributes ignoring `;` is a bug, but I am not inclined to make that argument

Comment (DanilaFe) [reply modules/standard/List.chpl:155]: It's a pleasant-surprise sort of bug -- or maybe a feature someone deliberately added? Anyway, nothing for you to do.

**Comment by bradcray** [comment]
PRs like this might suggest having more proactive discussions than we have been around whether there's agreement about which checks/fixits are obviously Chapeltastic vs. not.  Things like extraneous semicolons seem like no-brainers.  Things like choosing between `for 1..n` or `for i in 1..n` (where `i` is unused) might be more a matter of taste (not that I'm disagreeing with the change, which I happen to be a fan of; simply that I think it's worth getting buy-in on).

**Pull request merged**